lib/src/test/java/io/ably/lib/transport/ConnectionManagerTest.java

```diff
@@ -198,4 +198,35 @@ public void connectionmanager_reconnect_default_endpoint() throws AblyException
 		assertThat(connectionManager.getConnectionState().state, is(ConnectionState.connected));
 		assertThat(connectionManager.getHost(), is(equalTo(opts.realtimeHost)));
 	}
+
+	/**
+	 * <p>
+	 * Library and version param 'lib' should include the header value described there
+	 * {@link io.ably.lib.http.HttpUtils#X_ABLY_LIB_VALUE},
+	 * see {@link HttpHeaderTest#header_lib_channel_publish()}
+	 * </p>
+	 * <p>
+	 * Spec: RTN2g
+	 * </p>
+	 */
+	@Test
+	public void connectionmanager_param_lib() throws AblyException {
+		Setup.TestVars testVars = Setup.getTestVars();
+		ClientOptions opts = testVars.createOptions(testVars.keys[0].keyStr);
+		AblyRealtime ably = new AblyRealtime(opts);
+		ConnectionManager connectionManager = ably.connection.connectionManager;
+
+		new Helpers.ConnectionManagerWaiter(connectionManager).waitFor(ConnectionState.connected);
+
+		/* Get X-Ably-Lib param */
+		Param[] params = connectionManager.getConnectParams();
```

Comment:
Well yes if that is the best way.  The point is that we're testing internals hear, not behaviour.  So if you can think of any other way to test the behaviour of the connection then please do that, but failing that, a local test server should work?